### PR TITLE
Add isIFrameWindowInDOM function (closes #136)

### LINKED ISSUE #136: Tests that employ the focus functionality fail when run in several browsers
T285078

### PR DESCRIPTION
**Pull request by AndreyBelym**
\cc @inikulin @AlexanderMoskovkin

### CODE REVIEW

File: src/client/runner/runner-base.js [hunk 1]
```diff
@@ -45,7 +45,7 @@ function waitPageLoad (callback) {
     $(window).load(callbackWrapper);
     $(document).ready(function () {
         //NOTE: an iFrame may be removed in this moment
-        if (window && window.top)
+        if (testCafeCore.domUtils.isIFrameWindowInDOM(window))
```
Comment (AlexanderMoskovkin) [inline src/client/runner/runner-base.js]: `var domUtils` is defined yet in the top of the file, use it

**Comment by AlexanderMoskovkin** [comment]
tests failed, please fix

**Comment by AlexanderMoskovkin** [comment]
@AndreyBelym please rebase to the `master` branch to fix the tests (because of https://github.com/superroma/testcafe-phoenix/issues/153)

File: src/client/core/utils/dom.js [hunk 2]
```diff
@@ -323,6 +323,35 @@ export function isElementFocusable ($element) {
             $element.css('visibility') !== 'hidden');
 }
 
+export function isIFrameWindowInDOM (win) {
+    //NOTE: In MS Edge, if iframe is removed from DOM browser throws an exception when accessing window.top
+    //and window.frameElement. Fortunately, setTimeout is set to undefined in this case.
+    if (browserUtils.isMSEdge && !win.setTimeout)
```
Comment (inikulin) [inline src/client/core/utils/dom.js]: Just curios, can we get rid of browser detection here and just use the feature detection here?

**Comment by inikulin** [comment]
LGTM. Just have a question regarding feature detection. If it's not possible, merge as is.

File: src/client/core/utils/dom.js (hunk 2 above)
Comment (AndreyBelym) [inline src/client/core/utils/dom.js]: Yes, looks like we can use feature detection here.

**Comment by inikulin** [comment]
lgtm

File: test/client/fixtures/core/dom-utils-test.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    function finishTest () {
+        window.removeEventListener('message', onMessage, false);
+        start();
+    }
+
+    function onMessage (event) {
+        if (messageCounter === 0) {
+            equal(event.data, true);
+
+            var iFramePostMessage = $iFrame[0].contentWindow.postMessage.bind($iFrame[0].contentWindow);
```
Comment (AlexanderMoskovkin) [inline test/client/fixtures/core/dom-utils-test.js]: Add empty line after `var` declaration. Linting is not enabled here now but we will do it

**Comment by AlexanderMoskovkin** [comment]
/r- Just a minor style issue
@VasilyStrelyaev please, check the strings

File: src/client/core/utils/dom.js [hunk 4]
```diff
@@ -323,6 +323,35 @@ export function isElementFocusable ($element) {
             $element.css('visibility') !== 'hidden');
 }
 
+export function isIFrameWindowInDOM (win) {
+    //NOTE: In MS Edge, if iframe is removed from DOM browser throws an exception when accessing window.top
+    //and window.frameElement. Fortunately, setTimeout is set to undefined in this case.
```
Comment (VasilyStrelyaev) [inline src/client/core/utils/dom.js]:
```
    //NOTE: In MS Edge, if an iframe is removed from DOM, the browser throws an exception when accessing window.top
    //and window.frameElement. Fortunately, setTimeout is set to undefined in this case.
```

File: src/client/core/utils/dom.js [hunk 5]
```diff
@@ -323,6 +323,35 @@ export function isElementFocusable ($element) {
...
+    if (!win.setTimeout)
+        return false;
+
+    //NOTE: Cross-domain iframes in Firefox have null as frameElement even if they are in DOM
+    //But Firefox doesn't execute scripts in removed iframes, so we suppose that iframe is in DOM
```
Comment (VasilyStrelyaev) [inline src/client/core/utils/dom.js]:
```
    //NOTE: Cross-domain iframes in Firefox have null in frameElement even if they are in DOM
    //But Firefox doesn't execute scripts in removed iframes, so we suppose that the iframe is in DOM
```

**Comment by VasilyStrelyaev** [comment]
/r-
very-very minor articles

**Comment by VasilyStrelyaev** [comment]
lgtm

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Pull request merged**